### PR TITLE
Build: Use eslint-release

### PR DESCRIPTION
**Pull request by btmills**
Adapted [Makefile.js](https://github.com/eslint/typescript-eslint-parser/blob/65c2e0ae7b04c67d9bca17fb5a883cbe8a0deeaa/Makefile.js) from [typescript-eslint-parser](https://github.com/eslint/typescript-eslint-parser) and added [eslint-release](https://github.com/eslint/eslint-release) for automated releases.

### CODE REVIEW

File: Makefile.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+"use strict";
+
+/* eslint no-console: 0*/
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+require("shelljs/make");
+
+var nodeCLI = require("shelljs-nodecli");
```
Comment (ilyavolodin) [inline Makefile.js]: Do you need nodecli? `shelljs/make` already exposes `exec` command as is.

File: Makefile.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+// Helpers
+//------------------------------------------------------------------------------
+
+/**
+ * Generates a function that matches files with a particular extension.
+ * @param {string} extension The file extension (i.e. "js")
+ * @returns {Function} The function to pass into a filter method.
+ * @private
+ */
+function fileType(extension) {
```
Comment (ilyavolodin) [inline Makefile.js]: I know you just copied it from ESLint makefile, but you can use `ls('lib/**/*.js')` instead (I'm pretty sure you can, at least)

**Comment by not-an-aardvark** [comment]
What's the advantage of moving test/lint scripts to the Makefile rather than keeping them as npm scripts? At first glance it seems like unnecessary abstraction, but there is probably a reason that I'm missing.

File: Makefile.js (hunk 1 above)
Comment (btmills) [inline Makefile.js]: Looks like I was able to remove it!

File: Makefile.js (hunk 2 above)
Comment (btmills) [inline Makefile.js]: Indeed, that did work :+1:

**Comment by btmills** [comment]
> @not-an-aardvark What's the advantage of moving test/lint scripts to the Makefile rather than keeping them as npm scripts?

I'm going to go with "consistency with other eslint org projects", but I'm not attached to that argument. If the team would prefer npm scripts, I'm happy to do that too.

**Comment by not-an-aardvark** [comment]
My two cents is that using npm scripts might be better because additional flags can be provided, e.g. `npm run lint -- --fix` will pass the `--fix` flag to eslint. I don't feel strongly about this though, so I'm fine with whatever.

**Comment by btmills** [comment]
@not-an-aardvark just pushed up a commit to use npm scripts instead

**Pull request merged**